src/caret_analyze/architecture/struct/callback.py

```diff
@@ -0,0 +1,197 @@
+
+        Returns
+        -------
+        CallbackType
+            callback type
+
+        """
+        pass
+
+    @property
+    def callback_type_name(self) -> str:
+        return str(self.callback_type)
+
+    @property
+    def subscribe_topic_name(self) -> Optional[str]:
+        return self._subscribe_topic_name
+
+    @property
+    def publish_topic_names(self) -> Optional[Tuple[str, ...]]:
+        return self._publish_topic_names
+
+    @property
+    @abstractmethod
+    def summary(self) -> Summary:
+        pass
+
+    @abstractmethod
+    def to_value(self) -> CallbackStructValue:
+        pass
+
+
+class TimerCallbackStruct(CallbackStruct):
+    """Structured timer callback value."""
+
+    def __init__(
+        self,
+        node_name: str,
+        symbol: str,
+        period_ns: int,
+        publish_topic_names: Optional[Tuple[str, ...]],
+        callback_name: str,
+    ) -> None:
+        super().__init__(
+            node_name,
+            symbol,
+            None,
+            publish_topic_names,
+            callback_name)
+        self._period_ns = period_ns
+
+    @property
+    def callback_type(self) -> CallbackType:
+        return CallbackType.TIMER
+
+    @property
+    def period_ns(self) -> int:
+        return self._period_ns
+
+    @property
+    def summary(self) -> Summary:
+        return Summary({
+            'name': self.callback_name,
+            'type': self.callback_type_name,
+            'period_ns': self.period_ns
+        })
+
+    def to_value(self) -> TimerCallbackStructValue:
+        return TimerCallbackStructValue(
+            self.node_name, self.symbol, self.period_ns,
+            None if self.publish_topic_names is None else self.publish_topic_names,
+            self.callback_name)
+
+
+class SubscriptionCallbackStruct(CallbackStruct):
+    """Structured subscription callback value."""
+
+    def __init__(
+        self,
+        node_name: str,
+        symbol: str,
+        subscribe_topic_name: str,
+        publish_topic_names: Optional[Tuple[str, ...]],
+        callback_name: str,
+    ) -> None:
+        super().__init__(node_name, symbol, subscribe_topic_name,
+                         publish_topic_names, callback_name)
+        self.__subscribe_topic_name = subscribe_topic_name
+
+    @property
+    def callback_type(self) -> CallbackType:
+        return CallbackType.SUBSCRIPTION
+
+    @property
+    def subscribe_topic_name(self) -> str:
+        return self.__subscribe_topic_name
+
+    @property
+    def summary(self) -> Summary:
+        return Summary({
+            'name': self.callback_name,
+            'type': self.callback_type_name,
+            'topic': self.subscribe_topic_name
+        })
+
+    def to_value(self) -> SubscriptionCallbackStructValue:
+        return SubscriptionCallbackStructValue(
+            self.node_name, self.symbol,
+            '' if self.subscribe_topic_name is None else self.subscribe_topic_name,
```

Comment:
self.subscribe_topic_nameはstrになったので、冗長になっています。
```suggestion
            self.subscribe_topic_name,
```